crates/metaslang/bindings/src/lib.rs

```diff
@@ -22,23 +23,40 @@ type GraphHandle = stack_graphs::arena::Handle<stack_graphs::graph::Node>;
 pub struct Bindings<KT: KindTypes + 'static> {
     graph_builder_file: File<KT>,
     functions: Functions<KT>,
-    stack_graph: StackGraph,
+    pub stack_graph: StackGraph,
     cursors: HashMap<GraphHandle, Cursor<KT>>,
+    definiens: HashMap<GraphHandle, Cursor<KT>>,
+}
+
+pub trait PathResolver {
+    fn resolve_path(&self, context_path: &str, path_to_resolve: &str) -> Option<String>;
+}
+
+pub struct DefaultPathResolver;
+
+impl PathResolver for DefaultPathResolver {
+    fn resolve_path(&self, _context_path: &str, path_to_resolve: &str) -> Option<String> {
+        Some(path_to_resolve.to_string())
+    }
 }
 
 impl<KT: KindTypes + 'static> Bindings<KT> {
-    pub fn create(version: Version, binding_rules: &str) -> Self {
+    pub fn create(
+        version: Version,
+        binding_rules: &str,
+        path_resolver: Arc<dyn PathResolver + Sync + Send>,
```

Comment:
I wonder why are we using `Arc<>` here rather than `Rc<>` like everywhere else in the codebase? Is it required?

Reply:
This is a type restriction in [metaslang_graph_builder](https://github.com/manastech/slang/blob/feat%2Fimport-resolver/crates/metaslang/graph_builder/src/functions.rs#L89), which in turn is inherited from tree-sitter-graph. I don't mind the `Arc`, although since we're re-creating the functions for each stack graph, it's contained enough that I think we could change this to a `Box`.

OTOH, in the future we may want to, for example, parse and process several files in parallel and we would need to change this back. :shrug:

Reply:
Thanks for explaining. Let's keep it as is for now to not do additional work if not needed. We can always update it when we have the public final API.